ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -153,6 +153,30 @@ boolean controlCompactionForBlobStore(BlobStore store, boolean enable) {
     return compactionExecutor == null || compactionExecutor.controlCompactionForBlobStore(store, enable);
   }
 
+  /**
+   * Remove store from compaction manager.
+   * @param store the {@link BlobStore} to remove
+   * @return {@code true} if store is removed successfully. {@code false} if not.
+   */
+  boolean removeBlobStore(BlobStore store) {
+    boolean result;
+    if (compactionExecutor == null) {
+      stores.remove(store);
```

Comment:
Once #1226 is merged, `stores` will become concurrent set which should be ok here.